checks.d/spark.py

```diff
@@ -299,6 +307,33 @@ def _standalone_init(self, spark_master_address):
 
         return running_apps
 
+    def _mesos_init(self, master_address):
+        '''
+        Return a dictionary of {app_id: (app_name, tracking_url)} for running Spark applications.
+        '''
+        running_apps = {}
+
+        metrics_json = self._rest_request_to_json(master_address,
+            MESOS_MASTER_APP_PATH,
+            MESOS_SERVICE_CHECK)
+
+        if metrics_json.get('frameworks'):
+            for app_json in metrics_json.get('frameworks'):
```

Comment:
you could probably get rid of the `if` above and write this as `for app_json in metrics_json.get('frameworks', {}):` - if it's an empty dict we just won't iterate.